Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -366,6 +375,15 @@ protected void MuzzleFlash(EntityUid gun, AmmoComponent component, EntityUid? us
         CreateEffect(gun, ev, user);
     }
 
+    public void CauseImpulse(EntityCoordinates fromCoordinates, EntityCoordinates toCoordinates, PhysicsComponent userPhysics)
+    {
+        var fromMap = fromCoordinates.ToMap(EntityManager);
+        var toMap = toCoordinates.ToMapPos(EntityManager);
+        var mapDirection = toMap - fromMap.Position;
```

Comment:
its odd that these two, fromMap and toMap, arent both assigned the same type of variable
```suggestion
        var fromMap = fromCoordinates.ToMapPos(EntityManager);
        var toMap = toCoordinates.ToMapPos(EntityManager);
        var mapDirection = toMap - fromMap;
```

I suggest this change.

Also consider either renaming mapDirection (since its not really representative of a mapDirection, its the entire vector from the shooter to the coordinates where its trying to shoot) or normalizing the vector immediately upon assignment: `var mapDirection = (toMap - fromMap).Normalized;`

Reply:
yeah that way its much better